server/src/main/java/org/eclipse/lsp/cobol/core/preprocessor/delegates/ReplacePreProcessorListener.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (c) 2020 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+
+package org.eclipse.lsp.cobol.core.preprocessor.delegates;
+
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.antlr.v4.runtime.BufferedTokenStream;
+import org.antlr.v4.runtime.tree.TerminalNode;
+import org.apache.commons.lang3.tuple.Pair;
+import org.eclipse.lsp.cobol.core.CobolPreprocessor;
+import org.eclipse.lsp.cobol.core.CobolPreprocessorBaseListener;
+import org.eclipse.lsp.cobol.core.messages.MessageService;
+import org.eclipse.lsp.cobol.core.model.ExtendedDocument;
+import org.eclipse.lsp.cobol.core.model.Locality;
+import org.eclipse.lsp.cobol.core.model.ResultWithErrors;
+import org.eclipse.lsp.cobol.core.model.SyntaxError;
+import org.eclipse.lsp.cobol.core.preprocessor.delegates.util.PreprocessorStringUtils;
+import org.eclipse.lsp.cobol.core.preprocessor.delegates.util.ReplacingService;
+import org.eclipse.lsp.cobol.core.preprocessor.delegates.util.TokenUtils;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Deque;
+import java.util.List;
+
+import static org.antlr.v4.runtime.Token.EOF;
+import static org.eclipse.lsp.cobol.core.model.ErrorSeverity.ERROR;
+
+/**
+ * ANTLR listener, which deals only with the REPLACE compiler directives. So, that the rest of the
+ * pre-processing works on the replaced source.
+ */
+@Slf4j
+public class ReplacePreProcessorListener extends CobolPreprocessorBaseListener
+    implements GrammarPreprocessorListener {
+  @Getter private final List<SyntaxError> errors = new ArrayList<>();
+  private final ReplacingService replacingService;
+  private final List<Pair<String, String>> replacingClauses;
+  private final BufferedTokenStream tokens;
+  private final MessageService messageService;
+  private final String documentUri;
+  Deque<StringBuilder> textAccumulator = new ArrayDeque<>();
+
+  public ReplacePreProcessorListener(
+      ReplacingService replacingService,
+      MessageService messageService,
+      BufferedTokenStream tokens,
+      String documentUri,
+      List<Pair<String, String>> replacingClauses) {
+    this.replacingService = replacingService;
+    textAccumulator.push(new StringBuilder());
```

Comment:
Please, move this line after the fields initialization

Reply:
Done